### DISCUSSION TITLE
Esp 8266 , relay , switch on/off max

### DISCUSSION THREAD

**Discussion by lukaszwitkowski** [RavenSystem/esp-homekit-devices#1619] [MEPLHAA Scripts]
Hello ,
What is max relay and switch (on/off) not button we can connect to esp8266 i try 4 but i have only 3 working and i can't find another that working afrer reboot. 
When i use some pins for on/off whitch after reboot my esp/haa dont work.![image](https://user-images.githubusercontent.com/36861888/140897624-66858e24-505f-49f5-8c3f-f5a25bcef3be.png) photo from other project

**Comment by tonysprenk** [comment]
The photo is esp32 not esp8266. With MCP23017 Serial Interface Module you can add up to 248 extra GPIO.

**Reply by lukaszwitkowski** [reply]
Do you have any project / code json to work with that mcp? What gpio i write in json config?

**Comment by lukaszwitkowski** [comment]
I need to change source code ? Or do everything in web config? After flash by bin file?
